react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: '' };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ value: event.target.value });
+  }
+
+  handleSubmit(event) {
+    alert('An email was submitted: ' + this.state.value);
+    event.preventDefault();
+    console.log('State: ', this.state);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label> Email
+          <input type="text" value={this.state.value} onChange={this.handleChange}/>
+        </label>
+        <button>{'Sign Up'}</button>
```

Comment:
When using text in JSX it doesn't need to be within a JavaScript expression. It can just be:

```JSX
<button>Sign Up</button>
```

Reply:
Fixed!